src/Grpc.Net.Client/Balancer/Subchannel.cs

```diff
@@ -69,7 +69,7 @@ public sealed class Subchannel : IDisposable
     /// <summary>
     /// Gets the current connected address.
     /// </summary>
-    public BalancerAddress? CurrentAddress => _transport.CurrentAddress;
+    public BalancerAddress? CurrentAddress => GetAddress(_transport.CurrentEndPoint);
```

Comment:
how often is this invoked? i.e. is there a CPU or allocation problem here? (I notice that `GetAddress` is non-trivial); is this worth memoizing (remembering to wipe the memoized value if a relevant config change occurs)

Reply:
The reason I changed this to be lazily calculated is layering. It's hard to cache and get notifications when it changes. Fortunately, the property is rarely used. Load balancing is built around caching at a higher level.

The property is public API, so it can't change. If it could, then a method would be better, but the ship has sailed.